zhaquirks/tuya/ts0001_fingerbot.py

```diff
@@ -0,0 +1,145 @@
+from typing import Any, Dict, Optional, Union
+
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import Basic, Groups, Ota, Scenes, Time
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import TUYA_SEND_DATA
+from zhaquirks.tuya.mcu import (
+    DPToAttributeMapping,
+    EnchantedDevice,
+    TuyaMCUCluster,
+    TuyaOnOff,
+)
+
+
+class FingerBotMode(t.enum8):
+    CLICK = 0x00
+    SWITCH = 0x01
+    PROGRAM = 0x02
+
+
+class FingerBotReverse(t.enum8):
+    UP_ON = 0x00
+    UP_OFF = 0x01
+
+
+class TuyaFingerbotCluster(TuyaMCUCluster):
+    attributes = TuyaMCUCluster.attributes.copy()
+    attributes.update({101: ("mode", FingerBotMode)})
+    attributes.update({102: ("down_movement", t.uint16_t)})
+    attributes.update({103: ("sustain_time", t.uint16_t)})
+    attributes.update({104: ("reverse", FingerBotReverse)})
+    attributes.update({105: ("battery", t.uint16_t)})
+    attributes.update({106: ("up_movement", t.uint16_t)})
+    attributes.update({107: ("touch_control", t.Bool)})
```

Comment:
I think you should be able to use one `update` call with all attributes:

```suggestion
    attributes = TuyaMCUCluster.attributes.copy()
    attributes.update(
        {
            101: ("mode", FingerBotMode),
            102: ("down_movement", t.uint16_t),
            103: ("sustain_time", t.uint16_t),
            104: ("reverse", FingerBotReverse),
            105: ("battery", t.uint16_t),
            106: ("up_movement", t.uint16_t),
            107: ("touch_control", t.Bool),
        }
    )
```
(I know some Tuya code does it differently, but this seems a bit nicer.)